### PR TITLE
Java10 Hacking

### PR DESCRIPTION
**Pull request by svanoort**
* Handle Java version strings like "9" and "10" due to change in the nomenclature from 1.8, etc
    * Note that this requires all things taking up new plugin POM to change their java.level, i.e. "8" -> "1.8"

TODO:
* [ ] org.codehaus.mojo.signature artifactid hardcoded, because it does NOT include a period, i.e. 1.8 -> 18

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -521,7 +521,7 @@
                     <exclude>org.kohsuke.stapler:stapler-jelly</exclude>
                     <exclude>org.kohsuke.stapler:stapler-jrebel</exclude>
                     <exclude>org.jenkins-ci:task-reactor</exclude>
-                    <!--  findbugs dep managed to provided and optional so is not shipped and missing annotations ok -->
+                    <!--  findbugs dep managed to provided and optional so is not shipped and missing 3ations ok -->
```
Comment (KostyaSha) [inline pom.xml]: typo?

**Comment by KostyaSha** [comment]
will guice/guava/sezpoz as usual break? :D

**Comment by svanoort** [comment]
> will guice/guava/sezpoz as usual break? :D

Runs fine with Java 10 as of the latest builds AFAICT. 

The problem is all the tooling and so forth.

**Comment by svanoort** [comment]
@KostyaSha Runs okay with Java10 now, for the most part -- the tricky bit is it won't properly build/test in many cases with that JRE.

**Comment by oleg-nenashev** [comment]
@jenkinsci/java10-support

**Comment by KostyaSha** [comment]
> The problem is all the tooling and so forth.

Where is my popcorn! :)

**Comment by oleg-nenashev** [comment]
@svanoort maybe it makes sense to integrate with master (https://github.com/jenkinsci/plugin-pom/pull/114) and bump Animal Sniffer to 1.17

**Comment by svanoort** [comment]
@oleg-nenashev Done -- now to figure out how we deal with the org.codehaus.mojo.signature dependencies?

**Comment by svanoort** [comment]
@oleg-nenashev I agree those are better approaches -- I hadn't considered that approach, but it seems a lot better.

File: pom.xml [hunk 2]
```diff
@@ -628,7 +628,8 @@
         <configuration>
           <signature>
             <groupId>org.codehaus.mojo.signature</groupId>
-            <artifactId>java1${java.level}</artifactId>
+            <!-- Hardcoded unfortunately -->
```
Comment (daniel-beck) [inline pom.xml]: Unclear comment

**Comment by batmat** [comment]
Closing as per my last comment.

**Pull request closed without merging**